docs/extended_api/synchronization_library/pipeline_consumer_wait_prior.md

```diff
@@ -0,0 +1,52 @@
+---
+grand_parent: Extended API
+parent: Synchronization library
+---
+
+# cuda::**pipeline_consumer_wait_prior**
+
+Defined in header [`<cuda/pipeline>`](../headers/pipeline.md)
+
+```c++
+template<uint8_t Prior>
+void pipeline_consumer_wait_prior(pipeline<thread_scope_thread> & pipeline);
+```
+
+Blocks the current thread until all operations committed to _pipelines stages_ sequenced before the `Prior` last one complete. All stages up to `Prior` (excluded)
+are implicitly released.

```

Comment:
I get what this says, but probably only because I know what it means to say ;) How about:

"Let *Stage* be the pipeline stage `Prior` stages before the current one (counting the current one). Blocks the current thread until add operations committed to _pipeline stages_ up to *Stage* complete. All stages up to *Stage* (inclusive) are implicitly released." (Double check for off-by-ones in this wording.)

I'm trying to introduce an actual _stage_ that we can talk about, instead of talking about "`Prior` last one", which is a really confusing formulation when you don't already know what it means.

Reply:
Agreed, your version is better, I made two minor changes:
* until ~add~ **all** operations committed
* All stages up to Stage (~inclusive~ **exclusive**) are implicitly released.